src/ts/page/Signer.ts

```diff
@@ -1,30 +1,30 @@
-import { SignerOptions } from '@polkadot/api/types'
-import { InjectedSigner } from '@polkadot/extension-dapp/types'
-import { IExtrinsic } from '@polkadot/types/types'
+import { Signer as SignerInterface, SignerPayload, SignerResult } from '@polkadot/api/types'
 import { SendRequest } from './types'
 
 let sendRequest: SendRequest
+let nextId = 0
 
-export default class Signer implements InjectedSigner {
+export default class Signer implements SignerInterface {
   public constructor (_sendRequest: SendRequest) {
     sendRequest = _sendRequest
   }
 
-  public async sign (extrinsic: IExtrinsic, address: string,
-              { blockHash, blockNumber, era, genesisHash, nonce }: SignerOptions): Promise<number> {
-    // Bit of a hack - with this round-about way, we skip any keyring deps
-    const { id, signature } = await sendRequest('extrinsic.sign', JSON.parse(JSON.stringify({
-      address,
-      blockHash,
-      blockNumber: blockNumber.toNumber(),
-      era,
-      genesisHash,
-      method: extrinsic.method.toHex(),
-      nonce
-    })))
+  public async signPayload (payload: SignerPayload): Promise<SignerResult> {
+    const id = ++nextId
```

Comment:
should we use uuid here instead? do they have to be in sequence?

Reply:
Let's wait till the injectWeb3 api stable. If we do refactoring now, it'll be harder to sync Jaco's change later on. When he changes the api, usually a lot of files get touched. It would be easier to copy as a whole instead of piece by piece

Reply:
ok